src/Microsoft.Android.Build.BaseTasks/AsyncTaskExtensions.cs

```diff
@@ -79,28 +107,47 @@ public static ParallelLoopResult ParallelForEachWithLock<TSource> (this AsyncTas
 			});
 		}
 
-		static ParallelOptions ParallelOptions (AsyncTask asyncTask) => new ParallelOptions {
+		static ParallelOptions ParallelOptions (AsyncTask asyncTask, int maxConcurrencyLevel) => new ParallelOptions {
 			CancellationToken = asyncTask.CancellationToken,
-			TaskScheduler = TaskScheduler.Default,
+			TaskScheduler = GetTaskScheduler (maxConcurrencyLevel),
 		};
 
+		static TaskScheduler GetTaskScheduler (int maxConcurrencyLevel)
+		{
+			var pair = new ConcurrentExclusiveSchedulerPair (TaskScheduler.Default, maxConcurrencyLevel);
+			return pair.ConcurrentScheduler;
+		}
+
+		static int DefaultMaxConcurrencyLevel => Environment.ProcessorCount;
```

Comment:
We might want to use a similar bit of logic as aapt2 here. See https://github.com/xamarin/xamarin-android/blob/main/src/Xamarin.Android.Build.Tasks/Tasks/Aapt2.cs#L82. 

We have a MAX number of daemons of 6 and a minimum of 1.

Reply:
What if this was:
```suggestion
		static int DefaultMaxConcurrencyLevel => Math.Max (1, Environment.ProcessorCount - 1);
```

I think about AOT or other random tasks where we do `Parallel.ForEach()`, I'd think we'd want it to use more than 6 on my PC, for example. It has 16 cores.

`aapt2` is a little different because it's starting long running daemons. Where I think 6 would be sufficient on a many-core machine.